File: services/api-spring-boot/src/main/java/com/example/repository/ItemRepository.java
package com.example.repository;

import com.example.entity.Item;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

@Repository
public interface ItemRepository extends JpaRepository<Item, UUID> {

    List<Item> findByUser_Id(UUID userId);

    Optional<Item> findByIdAndUser_Id(UUID id, UUID userId);
}

File: services/18-spring-boot-kotlin/src/main/kotlin/com/example/repository/ItemRepository.kt
package com.example.repository

import com.example.entity.Item
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
interface ItemRepository : JpaRepository<Item, UUID> {
    fun findByUserId(userId: UUID): List<Item>
    fun findByIdAndUserId(id: UUID, userId: UUID): Item?
}
